src/psyclone/domain/gocean/transformations/kern_trans.py

```diff
@@ -69,16 +71,31 @@ class KernTrans(Transformation):
     ...     "  end subroutine\n"
     ...     "end module\n")
     >>> fortran_reader = FortranReader()
-    >>> kernel = fortran_reader.psyir_from_source(CODE)
-    >>> trans = kern_trans()
-    >>> trans.metadata_name = "compute_cu"
-    >>> trans.apply(kernel)
+    >>> kernel_container = fortran_reader.psyir_from_source(CODE)
+    >>> trans = kern_trans("compute_cu")
+    >>> trans.apply(kernel_container)
+
+    :param str metadata_name: the name of the symbol containing the \
+        required kernel metadata in language-level PSyIR.
+
+    :raises TransformationError: if the supplied metadata_name is \
+        invalid.
 
     '''
-    def __init__(self):
+    def __init__(self, metadata_name):
         super().__init__()
+
+        pattern = re.compile(r'[a-zA-Z_][\w]*')
+        if not metadata_name or not pattern.match(metadata_name):
+            raise TransformationError(
+                f"Error in {self.name} transformation. The kern_trans "
+                f"transformation requires the name of the variable "
+                f"containing the metadata to be set to a valid value "
```

Comment:
This needs rewording slightly now that it's in the constructor.

Reply:
updaterated.